codice/ddf
# [WIP] DDF-3613 As an administrator, I want configurations stored in Zookeeper so I can configure my cluster from a central location

#### What does this PR do?
Provides a proof of concept of Apache Felix Config Admin backed by Zookeeper. 

It's rough, lots of room for improvement, and does not have tests. But blueprint no longer blows up, and besides the standing up of the Zookeeper node(s) and a few system property changes, ALL admin tools (ui, command line, etc) are oblivious to the fact that Zookeeper is even being used. 

#### Who is reviewing it? 
Anyone interested. 

#### Choose 2 committers to review/merge the PR. 
Not quite a priority yet for tagging committers. 

#### How should this be tested?
Set up and **start** a Zookeeper `3.4.10` node or cluster. If defaults are used and both DDF and Zookeeper are on the same machine, no config tweaks are necessary. Otherwise, review the system properties. 

Start DDF normally. Run through the installer normally. 
Start **another** DDF normally. Run through the installer normally. 

Configure stuff using the UI. Replication should occur. Configure stuff by dropping files into `etc`. Replication should occur but be wary since the files themselves will not replicate. 

#### Any background context you want to provide?
TBD. 

#### What are the relevant tickets?
[DDF-3613](https://codice.atlassian.net/browse/DDF-3613)

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
